virtool/tasks/spawner.py

```diff
@@ -0,0 +1,92 @@
+import logging
+from typing import Type
+
+import aiojobs
+
+from virtool.config import Config
+from virtool.data.errors import ResourceError
+from virtool.data.utils import get_data_from_app
+from virtool.shutdown import (
+    shutdown_client,
+    shutdown_executors,
+    shutdown_scheduler,
+    shutdown_redis,
+    shutdown_authorization_client,
+)
+from virtool.startup import (
+    startup_events,
+    startup_http_client,
+    startup_databases,
+    startup_paths,
+    startup_executors,
+    startup_data,
+    startup_sentry,
+    startup_version,
+    startup_fake,
+    startup_fake_config,
+    startup_settings,
+)
+from virtool.tasks.task import BaseTask
+
+logger = logging.getLogger("task_spawner")
+
+
+async def create_app(config: Config):
+    """
+    Creates the Virtool application.
+
+    """
+
+    app = {"config": config, "mode": "task_spawner", "scheduler": aiojobs.Scheduler()}
+
+    on_startup = [
+        startup_version,
+        startup_http_client,
+        startup_fake_config,
+        startup_events,
+        startup_databases,
+        startup_paths,
+        startup_executors,
+        startup_fake,
+        startup_data,
+        startup_settings,
+        startup_sentry,
+    ]
+
+    for step in on_startup:
+        await step(app)
+
+    return app
+
+
+async def shutdown_app(app):
+    shutdown_steps = [
+        shutdown_authorization_client,
+        shutdown_client,
+        shutdown_executors,
+        shutdown_scheduler,
+        shutdown_redis,
+    ]
+
+    for step in shutdown_steps:
+        await step(app)
+
+
+def get_task_from_name(task_name: str) -> Type[BaseTask]:
+    matching_task = [cls for cls in BaseTask.__subclasses__() if cls.name == task_name]
+
+    if len(matching_task) != 1:
+        raise ResourceError("Invalid task name")
+
+    return matching_task[0]
+
+
+async def spawn(config: Config, task_name: str):
+    app = await create_app(config)
+
+    task = get_task_from_name(task_name)
+    logger.info("Spawning task %s", task.name)
+
+    await get_data_from_app(app).tasks.create(task)
```

Comment:
The above is a lot of work for this. Did you have any ideas on workarounds?

Reply:
I agree, and yes it is definitely possible but might be a pain from an upkeep perspective. 

The task spawner only really needs to A) register a task into the db and B) push the task id onto the redis list. Technically I could write up a bespoke solution to do that, but from an upkeep perspective I would way rather make use the `tasks` data_layer so that there is less things that need to be updated in the case that we change what it means to create a task.  This currently involves creating a bunch of connections and extra stuff that isn't really needed though

A good compromise could be to instantiate tasks data_layer on its own without the rest of the data_layers. This would mean that it only needs to connect to redis and pg which is pretty big on its own. Downside here is that if the tasks data_layer ever starts needing `open_fga`, `mongo` or other data_layers then this starts to make in some cases a lot less sense. I don't foresee tasks needing any of those resources though, so it is probably worth the risk.

Short version: Yes, I'll tweak it to only use the tasks data_layer and thereby greatly reduce the number of dependencies

Reply:
OK. I remember discussing this a bit already. You can leave as is if you want.